class_allocation.py

```diff
@@ -299,78 +301,66 @@ def ecpc(self):
         Equal Cumulative per Capita: Uses historical emissions, discount factors and
         population shares to allocate the global budget
         '''
-        # make sqrt curve
-        compensation_form_sqrt = np.sqrt(np.arange(0, 2101 - self.start_year_analysis))
-        # sum of values has to be 1
-        compensation_form_sqrt = compensation_form_sqrt / np.sum(compensation_form_sqrt)
-
-        xr_comp = xr.DataArray(compensation_form_sqrt, dims=['Time'], coords={'Time': self.analysis_timeframe})
-
         # Defining the timeframes for historical and future emissions
-        xrs = []
-        hist_emissions_startyears = self.settings['params']['hist_emissions_startyears']
-        discount_rates = self.settings['params']['discount_rates']
+        population_data = self.xr_total.Population.sel(Time=self.analysis_timeframe)
+        global_emissions_future = self.xr_total.GHG_globe.sel(Time=self.analysis_timeframe)
+        GF_frac = self.xr_total.GHG_hist.sel(Time=2021, Region=self.focus_region) / self.xr_total.GHG_hist.sel(Time=2021, Region='EARTH')
+        share_popt =  population_data / population_data.sel(Region='EARTH')
+        share_popt_past =  self.xr_total.Population / self.xr_total.Population.sel(Region='EARTH')
 
-        for startyear in hist_emissions_startyears:
+        xr_ecpc_all_list = []
+
+        for startyear in self.dim_histstartyear:
             hist_emissions_timeframe = np.arange(startyear, 1 + self.start_year_analysis)
-            future_emissions_timeframe = np.arange(self.start_year_analysis + 1, 2101)
 
             # Summing all historical CO2 emissions over the hist_emissions_timeframe
             hist_emissions = self.emis_hist.sel(Time = hist_emissions_timeframe)
 
             # Discounting -> We only do past discounting here
-            for discount in discount_rates:
-                past_timeline = np.arange(startyear, self.start_year_analysis + 1)
+            for discount in self.dim_discountrates:
 
+                # Add discounting stuff
+                past_timeline = np.arange(startyear, self.start_year_analysis + 1)
```

Comment:
You create this variable previously in line 314 with the name `hist_emissions_timeframe`. To avoid confusion it would be better to reuse the variable (it does not seem to be affected by the loop) or name it the same.